record/record.utils.js

```diff
@@ -628,14 +628,76 @@
+            // that is defined in google-dataset-config.js file
+            var catalogID = tuple.reference.location.catalogId, //getting the catalog-id without version
+                schemaName = tuple.reference.table.schema.name,
+                tableName = tuple.reference.table.name;
+            if (typeof googleDatasetConfigs !== 'undefined' &&
+                DataUtils.isObjectAndNotNull(googleDatasetConfigs) &&
+                DataUtils.isObjectAndNotNull(googleDatasetConfigs[catalogID]) &&
+                DataUtils.isObjectAndNotNull(googleDatasetConfigs[catalogID][schemaName]) &&
+                DataUtils.isObjectAndNotNull(googleDatasetConfigs[catalogID][schemaName][tableName])
+                ) {
+                var currConfig = googleDatasetConfigs[catalogID][schemaName][tableName];
+                if(currConfig) {
+                    var columns = currConfig.columns;
+                    var allValuesArray = currConfig.values;
+                    var valueInTuple;
+                    
+                    if (Array.isArray(columns) && Array.isArray(allValuesArray) && 
+                    (columns.length == allValuesArray.length || (columns.length == 1 && allValuesArray.length > 0))) {
+                        var isValuePresentInTuple = false;
+                        // 2D Array of values
+                        // { "columns": ["RID", "PID"], "values": [["RID_v1", "RID_v2"], ["PID_v1", "PID_v2"]]}
+                        if(Array.isArray(allValuesArray[0])) {
+                            columns.some(function(col, index) {
+                                valueInTuple = tuple.data[col];
+                                if (valueInTuple && allValuesArray[index].includes(valueInTuple)) {
+                                    isValuePresentInTuple = true;
+                                    return true;
+                                }
+                            });
+                        }
+                        // allowed case of 1D array of values
+                        // { "columns": ["RID"], "values": ["RID_v1", "RID_v2"]}
+                        else {
+                            valueInTuple = tuple.data[columns[0]];
+                            if (valueInTuple && allValuesArray.includes(valueInTuple)) {
+                                isValuePresentInTuple = true;
+                            }
+                        }
+                        if (!isValuePresentInTuple) {
+                            return;
+                        }
+                    }
+                }
+            }
+
+            // use ermrestjs and attach to dom if annotation exists
+            var metadata = $rootScope.reference.googleDatasetMetadata;
+            if (DataUtils.isObjectAndNotNull(metadata)) {
+                metadata = metadata.compute(tuple, $rootScope.templateVariables);
+                if (DataUtils.isObjectAndNotNull(metadata)) {
+                    metadata.url = window.location.href;
```

Comment:
Does this mean that data-modelers cannot define the `url`? Does url have to be the exact same value as window location? 
We need to discuss about this. We might want to use the resolver instead of window location.